### PR TITLE
feat: add site performance measuring event to analytics for `main`

### PR DESCRIPTION
**Pull request by danielbento92**
## Description

Add new event to analytics: Site Performance.
- Add compatibility to omnitracking;

### Dependencies

None.

## Checklist

- [x] The commit message follows our guidelines
- [x] Tests for the respective changes have been added
- [ ] The code is commented, particularly in hard-to-understand areas
- [x] The labels and/or milestones were added

## Disclaimer

By sending us your contributions, you are agreeing that your contribution is made subject to the terms of our [Contributor Ownership Statement](https://github.com/Farfetch/.github/blob/master/COS.md)

### CODE REVIEW

File: packages/analytics/src/integrations/Omnitracking/__tests__/Omnitracking.test.ts [hunk 1]
```diff
@@ -527,6 +527,44 @@ describe('Omnitracking', () => {
       );
     });
 
+    describe('Site Performance', () => {
+      it('should not track an Site Performance event if the required parameters are missing', async () => {
```
Comment (ruifcnunes) [inline packages/analytics/src/integrations/Omnitracking/__tests__/Omnitracking.test.ts:531]:
```suggestion
      it('should not track "Site Performance" event if the required parameters are missing', async () => {
```

File: packages/analytics/src/integrations/Omnitracking/__tests__/Omnitracking.test.ts [hunk 2]
```diff
@@ -527,6 +527,44 @@ describe('Omnitracking', () => {
...
+
+        expect(mockLoggerError).toHaveBeenCalledWith(
+          expect.stringContaining(
+            'to track this event, should be added a valid "performanceStats" property.',
+          ),
+        );
+        expect(postTrackingSpy).not.toHaveBeenCalled();
+      });
+
+      it('should not track an Site Performance event if the required parameters has wrong type', async () => {
```
Comment (ruifcnunes) [inline packages/analytics/src/integrations/Omnitracking/__tests__/Omnitracking.test.ts:549]: Same as above. Also, can we have a test with the happy path, where the correct value is passed?

File: packages/analytics/src/integrations/Omnitracking/definitions.ts [hunk 3]
```diff
@@ -764,6 +764,22 @@ export const trackEventsMapper: Readonly<OmnitrackingTrackEventsMapper> = {
...
+
+    if (performanceStats && typeof performanceStats === 'object') {
+      return {
+        tid: 1217,
+        performanceTimings: JSON.stringify(data.properties.performanceStats),
+      };
+    }
+
+    logger.error(
+      `[Omnitracking] - Event ${data.event}, to track this event, should be added a valid "performanceStats" property.`,
```
Comment (ruifcnunes) [inline packages/analytics/src/integrations/Omnitracking/definitions.ts:778]:
```suggestion
      `[Omnitracking] - Event "${data.event}": To track this event, should be added a valid "performanceStats" property.`,
```

File: packages/analytics/src/integrations/Omnitracking/__tests__/Omnitracking.test.ts (hunk 2 above)
Comment (danielbento92) [reply packages/analytics/src/integrations/Omnitracking/__tests__/Omnitracking.test.ts:549]: Done

File: packages/analytics/src/integrations/Omnitracking/__tests__/Omnitracking.test.ts (hunk 1 above)
Comment (danielbento92) [reply packages/analytics/src/integrations/Omnitracking/__tests__/Omnitracking.test.ts:531]: fixed

File: packages/analytics/src/integrations/Omnitracking/definitions.ts (hunk 3 above)
Comment (danielbento92) [reply packages/analytics/src/integrations/Omnitracking/definitions.ts:778]: done.

**Review by talbertosilva — APPROVED** [review]

File: packages/analytics/src/integrations/Omnitracking/__tests__/Omnitracking.test.ts [hunk 4]
```diff
@@ -527,6 +527,44 @@ describe('Omnitracking', () => {
...
+
+        expect(mockLoggerError).toHaveBeenCalledWith(
+          expect.stringContaining(
+            'track this event, should be added a valid "performanceStats" property.',
+          ),
+        );
+        expect(postTrackingSpy).not.toHaveBeenCalled();
+      });
+
+      it('should not track "Site Performance"event if the required parameters has wrong type', async () => {
```
Comment (boliveira) [inline packages/analytics/src/integrations/Omnitracking/__tests__/Omnitracking.test.ts:549]:
```suggestion
      it('should not track "Site Performance"event if the required parameters have the wrong type', async () => {
```

File: packages/analytics/src/integrations/Omnitracking/definitions.ts [hunk 5]
```diff
@@ -764,6 +764,22 @@ export const trackEventsMapper: Readonly<OmnitrackingTrackEventsMapper> = {
...
+
+    if (performanceStats && typeof performanceStats === 'object') {
+      return {
+        tid: 1217,
+        performanceTimings: JSON.stringify(data.properties.performanceStats),
+      };
+    }
+
+    logger.error(
+      `[Omnitracking] - Event "${data.event}": To track track this event, should be added a valid "performanceStats" property.`,
```
Comment (boliveira) [inline packages/analytics/src/integrations/Omnitracking/definitions.ts:778]:
```suggestion
      `[Omnitracking] - Event "${data.event}": To track this event, a valid "performanceStats" property should be added to the payload.`,
```

File: packages/analytics/src/integrations/Omnitracking/__tests__/Omnitracking.test.ts (hunk 4 above)
Comment (danielbento92) [reply packages/analytics/src/integrations/Omnitracking/__tests__/Omnitracking.test.ts]: Fixed.

File: packages/analytics/src/integrations/Omnitracking/definitions.ts (hunk 5 above)
Comment (danielbento92) [reply packages/analytics/src/integrations/Omnitracking/definitions.ts]: Fixed.

**Review by boliveira — APPROVED** [review]

**Pull request merged**